src/ServicePulse.Host/app/modules/monitoring/js/directives/ui.particular.largeGraph.js

```diff
@@ -163,23 +168,103 @@
                             }
 
                             var drawAverage = function(data, lineColor, fillColor) {
-                                drawAverageLine(chart, data, lineColor, fillColor, scaleX, scaleY);
+                                return drawAverageLine(chart, data, lineColor, fillColor, scaleX, scaleY);
                             }
-
+                            
+                            var displayAverageLabel = function(averageLine, label, value) {
+                                var {x , y, width} = averageLine.node().getBoundingClientRect();
+                                label.value(`${value.toFixed(2)}`);
+
+                                if(label.pointingToTheLeft){
+                                    label.displayAt(x + width, y);
+                                } else {
+                                    label.displayAt(x, y);
+                                }
+                                
+                            }
+                            
                             drawSeries(firstSeries, attrs.firstSeriesColor, attrs.firstSeriesFillColor);
 
                             if (secondSeries) {
                                 drawSeries(secondSeries, attrs.secondSeriesColor,attrs.secondSeriesFillColor);
                             }
 
-                            drawAverage(firstSeries, attrs.firstSeriesColor, attrs.firstSeriesFillColor );
+                            var firstAverageLine = drawAverage(firstSeries, attrs.firstSeriesColor, attrs.firstSeriesFillColor );
+
+                            var secondAverageLine = null;
 
                             if (secondSeries) {
-                                drawAverage(secondSeries, attrs.secondSeriesColor, attrs.secondSeriesFillColor);
+                                secondAverageLine = drawAverage(secondSeries, attrs.secondSeriesColor, attrs.secondSeriesFillColor);
                             }
+
+                            chart.on("mouseover", function() {                                
```

Comment:
@WojcikMike question: Do you think the closure functions I used here for the mouseover and mouseout events could create a memory leak? I am asking because the code of this ui.particular.largeGraph.js class redraws the whole SVG constantly by destroying all the elements in the SVG each time. 

scope.$watch('firstDataSeries', function is going to create a new chart variable every time it runs so in theory the chart instances from previous runs 


In the past to prevent possible memory leaks I created a function signature in the code sacrificing the readability but ensuring no new closure instances are created and kept referenced in the stack.

While checking the number of handlers register in the event using DevTools I see the number of registered closures remains constant, although I don't know if DevTools will refresh this in real-time:

<img width="699" alt="image" src="https://user-images.githubusercontent.com/545695/177712346-e70fcd2a-bb4b-4926-9c1e-ba02608b1f6c.png">

Reply:
The question sounds reasonable. And I would say that it should be causing memory leaks. I googled for a bit and found this (https://stackoverflow.com/questions/25059979/memory-release-of-d3-js-listener-inside-an-angularjs-directive). Som response suggest that modern browsers remove listener when an element is removed from DOM. Maybe that is why dev tools, do not show it.

Reply:
I ran version 1.31.3, and the code from this branch and the memory usage behavior remains identical.